### PR TITLE
FOV mouse sensitivity setting

### PR DESCRIPTION
**Pull request by SArpnt**
this adds a setting called FOV mouse sensitivity that adjusts how the mouse sensitivity behaves to fov changes

currently there's an system in place to try and keep screen movement distance equal with any fov, but this is more disorienting to some players than keeping the same amount of rotation (for example me)

at 1.0, the behavior is unchanged, at 0.0, it keeps the same rotations

a more common name for this would be ADS sensitivity (aim down sights) but this implies a shooter game, so i didn't go for this name.

## To do

This PR is Ready for Review.

- [ ] possibly change the name to something clearer
- [ ] fix some other code issues

## How to test

for checking whether mouse sensitivity here increases or decreases, measure the angles, for example by checking the mouse distance for a 360 rotation

- at any negative values, the sensitivity should decrease with higher fov
- at 0.0, the sensitivity should stay the exact same for all fov
- at 1.0, the sensitivity should behave exactly as it does originally
- at any positive value, the sensitivity should increase for higher fov

### CODE REVIEW

**Comment by SArpnt** [comment]
i just noticed i left in another change that modifies the main mouse sensitivity bounds.
i don't see it being harmful but it can be reverted if it isn't wanted

**Comment by sfan5** [comment]
Sounds useful to me but it definitely needs a better description if not better name, but I don't have a suggestion for that.

**Comment by SArpnt** [comment]
after thinking it over "ADS Mouse Sensitivity" is probably the best name since it's recognizable and people can figure out what it does otherwise by looking it up, even though it isn't technically the most accurate. maybe the description of it could take a disclaimer that it affects all zooming in and out. if a name gets approved i'll resolve the conflict and make the necessary changes

**Comment by rubenwardy** [comment]
If I didn't know what this was called in other programs, I'd say something like "Mouse sensitivity: FOV correction strength". But naming it the same as other games makes sense

File: src/client/game.cpp [hunk 1]
```diff
@@ -4150,7 +4158,7 @@ void Game::readSettings()
 
 	m_cache_fog_start = rangelim(m_cache_fog_start, 0.0f, 0.99f);
 	m_cache_cam_smoothing = rangelim(m_cache_cam_smoothing, 0.01f, 1.0f);
-	m_cache_mouse_sensitivity = rangelim(m_cache_mouse_sensitivity, 0.001, 100.0);
+	m_cache_mouse_sensitivity = rangelim(m_cache_mouse_sensitivity, 0.0f, std::numeric_limits<float>::max());
```
Comment (SmallJoker) [inline src/client/game.cpp:4161]: This change is nonsense. Even if you *ever* (rather never) need more than factor 100, use `std::max` instead.

Comment (SArpnt) [reply src/client/game.cpp:4161]: sorry, i made this fork a long time ago and i wasn't really considering a pull request, it should absolutely be updated

**Comment by SArpnt** [comment]
i'll leave the master branch to the last point i tested it at, and the quick-untested-merge branch just for resolving simple merge conflicts, i don't want to go far out of my way to maintain this for months just to see if some progress gets made.

not to be rude but i'm not sure why this hasn't already been reviewed or merged since as far as i can tell any issues have been resolved quite a while ago

**Comment by Zughy** [comment]
@SArpnt rebase needed, I'm sorry

**Comment by SArpnt** [comment]
> @SArpnt rebase needed, I'm sorry

i get the feeling that if i do that, a few months later i'm going to be told this needs another rebase.
why is this something i need to do, and not something that someone on the project can do when they're actually about to merge it?
that's not a rhetorical question, i genuinely don't get why a merge doesn't work or why nobody else is able to rebase.
i have issues with cloning and pushing to github right now and i can't figure out any way to check what i need to in the web ui, i can't work on this for a while and this feels like something that anyone else could do in 2 minutes

**Comment by Zughy** [comment]
> i genuinely don't get why a merge doesn't work or why nobody else is able to rebase

because code on master has been changed and because if we rebased all the PRs that actually need a rebase, we'd end up with even less time than we currently have (we're all volunteers). This also answers to why nobody has checked your PR yet: we're not evil, we're just few :( I could put it into the 5.7 milestone but it's not really my call, I'm sorry. What I can do is remind core devs to review it; just, I need the rebase first

**Comment by SArpnt** [comment]
i changed the head branch of this pull request before when i did the quick-untested-merge branch, but now i can't find any way to change it again or any evidence that it was even possible?
i'll create a new pr with the improved branch

**Pull request closed without merging**